api/apps/api/src/modules/access-control/scenarios-acl/scenario-acl.service.ts

```diff
@@ -161,6 +180,89 @@ export class ScenarioAclService implements ScenarioAccessControl {
     return otherOwnersInScenario >= 1;
   }
 
+  async acquireLock(
+    userId: string,
+    scenarioId: string,
+  ): Promise<Either<typeof forbiddenError | AcquireFailure, ScenarioLockDto>> {
+    if (!(await this.canEditScenario(userId, scenarioId))) {
+      return left(forbiddenError);
+    }
+    const result = await this.lockService.acquireLock(scenarioId, userId);
+    if (isLeft(result)) {
+      return result;
+    }
+    return result;
```

Comment:
??

Reply:
The underneath method returns a monad, so in the first case result will be a left object and in the second case it will be a right object. It did confuse me at the beginning but I got so used to it that now I feel nothing.